src/solver/conv_hip_implicit_gemm_bwd_v4r1.cpp

```diff
@@ -218,7 +218,7 @@ PerformanceImplicitGemmBwdDataV4R1::CalculateGemmABlockCopyPerformanceParameters
         // decide threadwise copy lengths
         const auto a_data_per_thread_copy_gemmm = SrcDataPerRead_GemmM;
         const auto a_data_per_thread_copy_gemmk =
-            a_data_per_thread_copy / a_data_per_thread_copy_gemmm;
+            a_data_per_thread_copy / a_data_per_thread_copy_gemmm; // NOLINT
```

Comment:
It was not necessary before your PR, why it is necessary now? Does clang-tidy on CI blame this?

Reply:
http://micimaster.amd.com/blue/rest/organizations/jenkins/pipelines/MLLibs/pipelines/MIOpen/branches/cderb%252Ffin_perf_solver_mergres2/runs/38/nodes/13/steps/87/log/?start=0
yes

Reply:
[Informative] @cderb I can't access CI (micimaster.amd.com)

Reply:
> [Informative] @cderb I can't access CI (micimaster.amd.com)

Oh, Sorry!
```
[2022-06-08T18:16:45.191Z] /var/jenkins/workspace/n_cderb_fin_perf_solver_mergres2/src/solver/conv_hip_implicit_gemm_bwd_v4r1.cpp:221:36: error: Division by zero [clang-analyzer-core.DivideZero,-warnings-as-errors]
[2022-06-08T18:16:45.191Z]             a_data_per_thread_copy / a_data_per_thread_copy_gemmm;
```
``` 
[2022-06-08T18:16:49.973Z] /var/jenkins/workspace/n_cderb_fin_perf_solver_mergres2/src/solver/conv_hip_implicit_gemm_fwd_v4r4.cpp:319:46: error: The result of the '/' expression is undefined [clang-analyzer-core.UndefinedBinaryOperatorResult,-warnings-as-errors]
[2022-06-08T18:16:49.973Z]         ClusterLengths_GemmK = GemmKPerBlock / b_data_per_thread_copy_gemmk;
```
which was just this prior commit [9563a5e](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1548/commits/9563a5e49ae0f3ed218de11168ce6b33a90bd71f)

Reply:
I'll branch it and see of removing the nolints would work.

Reply:
Confirmed that after merge with develop, removing the nolint will still result in a static analyzer error.
Note that this error started after making functions in generic_search callable from any_solver.

Reply:
Ok

Reply:
[Recommendation] But I would prefer in this case replacing NOLINT with
```C++
if(b_data_per_thread_copy_gemmk == 0)
   MIOPEN_THROW("DIV/0 with b_data_per_thread_copy_gemmk");
```
Up to you.

Reply:
Resolved!